### PR TITLE
Add CodeChef presences 

### PR DESCRIPTION
**Pull request by AvinashDhillor**
Proof:

<details>
  <summary>discuss.codechef.com</summary>

### Discussion home
discuss
  
### Discussion topic
discuss topic
</details>

<details>
  <summary>blog.codechef.com</summary>

### Blog home

blog home

### Blog topic
blog topic

### Author profile
blog author profile

</details>

<details>
  <summary>Home</summary>

home

</details>
<details>
  <summary>Problem by tags</summary>

tags

</details>
<details>
  <summary>Problems</summary>

problems

</details>
<details>
  <summary>IDE</summary>

ide

</details>
<details>
  <summary>AIMICPC</summary>

aimicpc

</details>
<details>
  <summary>Contests</summary>

contests

</details>
<details>
  <summary>Certification</summary>

certification

</details>
<details>
  <summary>Careers</summary>

carrers

</details>
<details>
  <summary>Signup</summary>

sign up

</details>
<details>
  <summary>Wiki</summary>

wiki

</details>
<details>
  <summary>Users</summary>

users

</details>

### CODE REVIEW

File: websites/C/CodeChef/presence.ts [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+    clientId: "845716323296083999"
+  }),
+  timeElapsed: number = ~~Date.now() / 1000;
+
+presence.on("UpdateData", async () => {
+  const presenceData: PresenceData = {
+      largeImageKey: "logo",
+      startTimestamp: timeElapsed
+    },
+    domain = location.host;
```
Comment (ririxidev) [inline websites/C/CodeChef/presence.ts:11]:
```suggestion
    domain = location.host.split(".");
```

File: websites/C/CodeChef/presence.ts [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+
+  if (domain.includes("blog")) {
```
Comment (ririxidev) [inline websites/C/CodeChef/presence.ts:13]: ``switch(domain[0])`` :)
please change it to this

File: websites/C/CodeChef/presence.ts [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+    presenceData.details = "Viewing:";
+    presenceData.state = "Problems By Tag Name";
+  } else if (location.pathname.split("/").includes("problems")) {
+    const contestName: string =
+        document.querySelector(".breadcrumbs").lastElementChild.textContent,
+      problemName: string = document.title.split("|")[0];
+    presenceData.details = contestName;
+    presenceData.state = "Solving: " + problemName;
+    presenceData.smallImageKey = "code";
+    presenceData.smallImageText = "coding";
```
Comment (ririxidev) [inline websites/C/CodeChef/presence.ts:54]:
```suggestion
```

File: websites/C/CodeChef/presence.ts [hunk 4]
```diff
@@ -0,0 +1,127 @@
...
+  } else if (domain.includes("discuss")) {
+    // ? https://discuss.codechef.com
+    presenceData.details = "Viewing Discussions";
+    if (location.pathname.split("/")[1] === "t") {
+      const discussionTopicHeading: string =
+        document.querySelector(".fancy-title").textContent;
+      presenceData.details = "Reading Discussions:";
+      presenceData.state = discussionTopicHeading;
+      presenceData.smallImageKey = "reading";
+      presenceData.smallImageText = "Reading";
```
Comment (ririxidev) [inline websites/C/CodeChef/presence.ts:40]:
```suggestion
```

File: websites/C/CodeChef/presence.ts [hunk 5]
```diff
@@ -0,0 +1,127 @@
...
+    else if (location.pathname.split("/")[1] === "author") {
+      presenceData.details = "Viewing Profile: ";
+      presenceData.state = "Author: " + location.pathname.split("/")[2];
+    } else if (location.pathname.split("/")[4]) {
+      presenceData.details =
+        "Reading: " + document.querySelector(".posttitle").textContent;
+      presenceData.state =
+        "Author: " + document.querySelector(".author-username").textContent;
+      presenceData.smallImageKey = "reading";
+      presenceData.smallImageText = "Reading";
```
Comment (ririxidev) [inline websites/C/CodeChef/presence.ts:29]:
```suggestion
```

File: websites/C/CodeChef/presence.ts [hunk 6]
```diff
@@ -0,0 +1,127 @@
...
+      largeImageKey: "logo",
+      startTimestamp: timeElapsed
+    },
+    domain = location.host;
+
+  if (domain.includes("blog")) {
+    // ? https://blog.codechef.com
+    presenceData.details = "Viewing Blogs";
+    presenceData.smallImageKey = "reading";
+    presenceData.smallImageText = "Reading";
```
Comment (ririxidev) [inline websites/C/CodeChef/presence.ts:17]:
```suggestion
```

File: websites/C/CodeChef/presence.ts [hunk 7]
```diff
@@ -0,0 +1,127 @@
...
+  } else {
+    // ? https://codechef.com
+    switch (location.pathname.split("/")[1]) {
+      case "ide": {
+        const compilerName: string =
+          document.querySelector(".chosen-single").childNodes[0].textContent;
+        presenceData.details = "Using IDE";
+        presenceData.state = "Editing " + compilerName + " file";
+        presenceData.smallImageKey = "code";
+        presenceData.smallImageText = "coding";
```
Comment (ririxidev) [inline websites/C/CodeChef/presence.ts:64]:
```suggestion
```

File: websites/C/CodeChef/dist/metadata.json [hunk 8]
```diff
@@ -0,0 +1,27 @@
+{
+    "$schema": "https://schemas.premid.app/metadata/1.3",
```
Comment (ririxidev) [inline websites/C/CodeChef/dist/metadata.json:2]:
```suggestion
    "$schema": "https://schemas.premid.app/metadata/1.4",
```

**Review by ririxidev — CHANGES_REQUESTED** [review]
Please check my review 😃

**Comment by ririxidev** [comment]
umm sorry for that but could you update your fork because when I was doing pr I broke something 😓

**Comment by AvinashDhillor** [comment]
okay

**Review by ririxidev — APPROVED** [review]
thanks :)

**Review by Bas950 — APPROVED** [review]

**Pull request merged**